LDMP/models/algorithms_delegate.py

```diff
@@ -0,0 +1,263 @@
+"""
+
+__author__ = 'Luigi Pirelli / Kartoza'
+__date__ = '2021-03-03'
+
+from functools import partial
+from typing import Optional, Union
+from qgis.PyQt.QtCore import (
+    QModelIndex,
+    Qt,
+    QCoreApplication,
+    QObject,
+    pyqtSignal,
+    QRectF,
+    QAbstractItemModel
+)
+from qgis.PyQt.QtWidgets import (
+    QStyleOptionViewItem,
+    QToolButton,
+    QMenu,
+    QStyledItemDelegate,
+    QItemDelegate,
+    QWidget,
+    QAction
+)
+from qgis.PyQt.QtGui import (
+    QPainter,
+    QPixmap
+)
+from qgis.PyQt.QtSvg import (
+    QSvgRenderer
+)
+
+from LDMP.models.algorithms import (
+    AlgorithmGroup,
+    AlgorithmDescriptor,
+    AlgorithmNodeType,
+    AlgorithmRunMode
+)
+from LDMP import __version__, log, tr
+
+
+# Icons used for animation:
+# inspired by https://wiki.python.org/moin/PyQt/Animated%20items%20using%20delegates
+_icon0_xpm = [
+    "16 16 4 1",
+    ".  c None",
+    "s  c #cccc00",
+    "x  c #ffffff",
+    "   c #000000",
+    "................",
+    ".              .",
+    ". xxxxxxxxxxxx .",
+    ". x          x .",
+    ".. x ssssss x ..",
+    "... x ssss x ...",
+    ".... x ss x ....",
+    "..... x  x .....",
+    "..... x  x .....",
+    ".... x    x ....",
+    "... x      x ...",
+    ".. x        x ..",
+    ". x          x .",
+    ". xxxxxxxxxxxx .",
+    ".              .",
+    "................"]
+
+_icon1_xpm = [
+    "16 16 4 1",
+    ".  c None",
+    "s  c #cccc00",
+    "x  c #ffffff",
+    "   c #000000",
+    "................",
+    ".              .",
+    ". xxxxxxxxxxxx .",
+    ". x          x .",
+    ".. x s    s x ..",
+    "... x ssss x ...",
+    ".... x ss x ....",
+    "..... x  x .....",
+    "..... x  x .....",
+    ".... x ss x ....",
+    "... x  ss  x ...",
+    ".. x        x ..",
+    ". x          x .",
+    ". xxxxxxxxxxxx .",
+    ".              .",
+    "................"]
+
+_icon2_xpm = [
+    "16 16 4 1",
+    ".  c None",
+    "s  c #cccc00",
+    "x  c #ffffff",
+    "   c #000000",
+    "................",
+    ".              .",
+    ". xxxxxxxxxxxx .",
+    ". x          x .",
+    ".. x        x ..",
+    "... x ssss x ...",
+    ".... x ss x ....",
+    "..... x  x .....",
+    "..... x  x .....",
+    ".... x ss x ....",
+    "... x  ss  x ...",
+    ".. x   ss   x ..",
+    ". x          x .",
+    ". xxxxxxxxxxxx .",
+    ".              .",
+    "................"]
+
+_icon3_xpm = [
+    "16 16 4 1",
+    ".  c None",
+    "s  c #cccc00",
+    "x  c #ffffff",
+    "   c #000000",
+    "................",
+    ".              .",
+    ". xxxxxxxxxxxx .",
+    ". x          x .",
+    ".. x        x ..",
+    "... x      x ...",
+    ".... x    x ....",
+    "..... x  x .....",
+    "..... x  x .....",
+    ".... x ss x ....",
+    "... x ssss x ...",
+    ".. x ssssss x ..",
+    ". x          x .",
+    ". xxxxxxxxxxxx .",
+    ".              .",
+    "................"]
```

Comment:
As much as I appreciate this ASCII art I don't think we should be adding icons to the algorithms tab, at least not at this early stage of development. This was not part of our original design, so we are getting into more trouble than initially foreseen with this.

Reply:
ok it will be removed...